docs/rfcs/0007-flow-modules.md

```diff
@@ -0,0 +1,295 @@
+
+*Pros*
+
+* Allows more resilient usage
+
+*Cons*
+
+* Can create undefined behavior
+* Complex to unload and reload
+
+## `modules.multiple`: Allowing multiple modules to be loaded at once
+
+Note: This feels the most experimental of the topics listed.
+
+Exports are accessed via `module.multiple.LABEL.exports.NAME`, `exports` is a `map(array)`. The `exports` label is used to prevent any collision if other fields are added to `modules.multiple`
+
+## Option 1: No filter 
+
+Depend on filtering the `map(string)` input before loading the modules.
+
+
+#### Filter prior
+
+Assume all files in `/configs/*river` are appropriate for the `module.multiple.load`. This has the advantage of being simplifying the usage of modules and development. This also mirrors `module.single` in the being simple and pushing verisoning/usage to other components. Downside is users are unable to put all configs in one folder and magically work.
+
+```river
+
+// Note this doesnt exist and should only be used for representative purposes.
+local.files "loadfolder" {
+    folder_path = "/configs" # Assume this outputs a map(string)
+    filter = "*.river"
+}
+
+module.multiple "load" {
+    source = local.files.loadfolder.contents
+}
+
+prometheus.scrape "module" {
+    // The module.multiple coalesces multiple exports into an array. 
+    targets = module.multiple.load.exports.targets
+}
+
+```
+
+## Option 2: Filter
+
+Allow `filter_arguments` and `filter_exports` to only include modules that define arguments and exports in the filters. Modules may have additional arguments that can optionally be set.
+
+
+
+#### Filter by arguments and exports
+
+Pass all files to both `load` and `load2` filtering if they have the appropriate input. This can lead to duplicate loading of modules if a module defines both `input1` and `input2`.
+
+```river
+
+// Note this doesnt exist and should only be used for representative purposes.
+local.files "loadfolder" {
+    folder_path = "/configs" # Assume this outputs a map(string)
+    filter = "*.river"
+}
+
+module.multiple "load" {
+    source = local.files.loadfolder.contents
+    filter = ["input1"]
+}
+
+module.multiple "load2" {
+    source = local.files.loadfolder.contents
+    filter = ["input2"]
+}
+
+```
+
+
+# Example Documentation for `argument`
+
+## Arguments
+
+The following arguments are supported:
+
+Name            | Type                | Description                                                                                | Default | Required
+--------------- | ------------------- | ------------------------------------------------------------------------------------------ |---------| --------
+`optional`  | `bool` | If an argument has to be specified. |    "false"     | no
```

Comment:
I think we talked about this 1 and using the false as default which makes sense. If we decide it makes sense for the default of `optional` to be true we could always rename this to `required` default false.

Reply:
Absolutely.